pyop2/fusion/interface.py

```diff
@@ -0,0 +1,331 @@
+# This file is part of PyOP2
+#
+# PyOP2 is Copyright (c) 2016, Imperial College London and
+# others. Please see the AUTHORS file in the main source directory for
+# a full list of copyright holders.  All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in the
+#       documentation and/or other materials provided with the distribution.
+#     * The name of Imperial College London or that of other
+#       contributors may not be used to endorse or promote products
+#       derived from this software without specific prior written
+#       permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTERS
+# ''AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDERS OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+# OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""Interface for loop fusion. Some functions will be called from within PyOP2
+itself, whereas others directly from application code."""
+
+import os
+import sys
+from contextlib import contextmanager
+from decorator import decorator
+
+from pyop2.base import _LazyMatOp
+from pyop2.mpi import MPI
+from pyop2.logger import warning, info as log_info
+from pyop2.utils import flatten
+
+try:
+    """Is SLOPE accessible ?"""
+    sys.path.append(os.path.join(os.environ['SLOPE_DIR'], 'python'))
+    import slope_python as slope
+
+    # Set the SLOPE backend
+    backend = os.environ.get('SLOPE_BACKEND')
+    if backend not in ['SEQUENTIAL', 'OMP']:
+        backend = 'SEQUENTIAL'
+    if MPI.COMM_WORLD.size > 1:
```

Comment:
Mmm I see...what's the PyOP2 way of checking whether I'm running with more than one process ?

Previously I could use ``MPI.parallel``